provider/osio/osio.go

```diff
@@ -0,0 +1,119 @@
+package osio
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/containous/traefik/log"
+	"github.com/containous/traefik/provider"
+	"github.com/containous/traefik/safe"
+	"github.com/containous/traefik/types"
+)
+
+// Provider holds configurations of the provider.
+type Provider struct {
+	provider.BaseProvider `mapstructure:",squash" export:"true"`
+	ClustersAPI           string `description:"URL to get Clusters data" export:"true"`
+}
+
+type cluster struct {
+	APIURL     string `json:"api-url"`
+	AppDNS     string `json:"app-dns"`
+	ConsoleURL string `json:"console-url"`
+	MetricsURL string `json:"metrics-url"`
+	Name       string `json:"name"`
+}
+
+type clustersAPIResponse struct {
+	Clusters []cluster `json:"data"`
+}
+
+// Provide allows the osio provider to provide configurations to traefik
+// using the given configuration channel.
+func (p *Provider) Provide(configChan chan<- types.ConfigMessage, pool *safe.Pool, constraints types.Constraints) error {
+	config, err := p.loadConfig()
+	if err != nil {
+		return err
+	}
+	go p.runPeriodically(configChan)
+	sendConfigToChannel(configChan, config)
+	return nil
+}
+
+func (p *Provider) loadConfig() (*types.Configuration, error) {
+	clustersAPIResponse := callClustersAPI(p.ClustersAPI)
+	return loadRules(clustersAPIResponse)
+}
+
+func (p *Provider) runPeriodically(configChan chan<- types.ConfigMessage) {
+	for {
+		time.Sleep(5 * time.Second)
+		p.checkConfig(configChan)
```

Comment:
Check config on regular interval.  Kept 5 sec but it will be ~15 mins in final version.